src/ORM/Association/HasMany.php

```diff
@@ -182,6 +182,175 @@ public function saveAssociated(EntityInterface $entity, array $options = [])
+     * of this association
+     * @param array $targetEntities list of entities belonging to the `target` side
+     * of this association
+     * @param array $options list of options to be passed to the save method
+     * @return bool true on success, false otherwise
+     */
+    public function link(EntityInterface $sourceEntity, array $targetEntities, array $options = [])
+    {
+        $saveStrategy = $this->saveStrategy();
+        $this->saveStrategy(self::SAVE_APPEND);
+        $source = $this->source();
+        $property = $this->property();
+
+        $currentEntities = (new Collection((array)$sourceEntity->get($property)))->append($targetEntities);
+
+        $sourceEntity->set($property, array_unique($currentEntities->toList()));
+
+        $savedEntity = $this->saveAssociated($sourceEntity);
+
+        $ok = ($savedEntity instanceof EntityInterface);
+
+        $this->saveStrategy($saveStrategy);
+
+        if ($ok) {
+            $sourceEntity->set($property, $savedEntity->get($property));
+            $sourceEntity->clean($property);
+        }
+
+        return $ok;
+    }
+
+    /**
+     * Removes all links between the passed source entity and each of the provided
+     * target entities. This method assumes that all passed objects are already persisted
+     * in the database and that each of them contain a primary key value.
+     *
+     * By default this method will also unset each of the entity objects stored inside
+     * the source entity.
+     *
+     * ### Example:
+     *
+     * ```
+     * $user = $users->get(1);
+     * $user->articles = [$article1, $article2, $article3, $article4];
+     * $users->save($user, ['Associated' => ['Articles']]);
+     * $allArticles = [$article1, $article2, $article3];
+     * $users->Articles->unlink($user, $allArticles);
+     * ```
+     *
+     * `$article->get('articles')` will contain only `[$article4]` after deleting in the database
+     *
+     * @param \Cake\Datasource\EntityInterface $sourceEntity an entity persisted in the source table for
+     * this association
+     * @param array $targetEntities list of entities persisted in the target table for
+     * this association
+     * @param bool $cleanProperty whether or not to remove all the objects in $targetEntities
+     * that are stored in $sourceEntity
+     * @throws \InvalidArgumentException if non persisted entities are passed or if
+     * any of them is lacking a primary key value
+     * @return void
+     */
+    public function unlink(EntityInterface $sourceEntity, array $targetEntities, $cleanProperty = true)
+    {
+        $foreignKey = (array)$this->foreignKey();
+        $target = $this->target();
+        $targetPrimaryKey = array_merge((array)$target->primaryKey(), (array)$foreignKey);
+        $property = $this->property();
+
+        $conditions = [
+        'OR' => (new Collection($targetEntities) )
```

Comment:
I wonder why PHPCS didn't catch this. I have fixed it in my last commit. Thank you